### PR TITLE
MINOR: Update upgrade guide for ZK

### PR DESCRIPTION
**Pull request by bbejeck**
Adding the ZK upgrade information to the `Notable Changes` section for 2.4.1

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
ping @rondagostino, @ijuma, and @JimGalasyn

File: docs/upgrade.html [hunk 1]
```diff
@@ -19,6 +19,17 @@
 
 <script id="upgrade-template" type="text/x-handlebars-template">
 
+<h5><a id="upgrade_241_notable" href="#upgrade_241_notable">Notable changes in 2.4.1</a></h5>
```
Comment (rondagostino) [inline docs/upgrade.html]: I think this needs to go down below the `upgrade_2_4_0` section.  See, for example, `upgrade_221_notable` and `upgrade_111_notable` sections, which appear below the `upgrade_2_2_0` and `upgrade_1_1_0` sections, respectively.

**Review by rondagostino — COMMENTED** [review]
Not sure if anything else is notable besides the ZK upgrade, but the ZK 3.5.7 upgrade note (and anything else, if there is anything else) I think needs to move down a section.

File: docs/upgrade.html (hunk 1 above)
Comment (bbejeck) [reply docs/upgrade.html]: Ack.  That's where I had it initially, but I decided to put it up on top.  I'll revert and move it back to where it should be.  Thanks for the review!

**Comment by bbejeck** [comment]
ping @rondagostino, @ijuma, and @JimGalasyn

File: docs/upgrade.html [hunk 2]
```diff
@@ -82,6 +82,17 @@ <h4><a id="upgrade_2_4_0" href="#upgrade_2_4_0">Upgrading from 0.8.x, 0.9.x, 0.1
     </li>
 </ol>
 
+<h5><a id="upgrade_241_notable" href="#upgrade_241_notable">Notable changes in 2.4.1</a></h5>
```
Comment (bbejeck) [inline docs/upgrade.html:85]: From what I can see in the file, the `notable` notes for the current release go before the ones for the previous release.

File: docs/upgrade.html [hunk 3]
```diff
@@ -82,6 +82,17 @@ <h4><a id="upgrade_2_4_0" href="#upgrade_2_4_0">Upgrading from 0.8.x, 0.9.x, 0.1
...
+<ul>
+    <li>
+        ZooKeeper has been upgraded to 3.5.7, and a ZooKeeper upgrade from 3.4.X to 3.5.7 can fail if there are no snapshot files in the 3.4 data directory.
+        This usually happens in test upgrades where ZooKeeper 3.5.7 is trying to load an existing 3.4 data dir in which no snapshot file has been created.
+        For more details about the issue please refer to <a href="https://issues.apache.org/jira/browse/ZOOKEEPER-3056">ZOOKEEPER-3056</a>.
```
Comment (JimGalasyn) [inline docs/upgrade.html]:
```suggestion
        For more information, see <a href="https://issues.apache.org/jira/browse/ZOOKEEPER-3056">ZOOKEEPER-3056</a>.
```

File: docs/upgrade.html [hunk 4]
```diff
@@ -82,6 +82,17 @@ <h4><a id="upgrade_2_4_0" href="#upgrade_2_4_0">Upgrading from 0.8.x, 0.9.x, 0.1
...
+        A fix is given in <a href="https://issues.apache.org/jira/browse/ZOOKEEPER-3056">ZOOKEEPER-3056</a>, which is to set <code>snapshot.trust.empty=true</code>
```
Comment (JimGalasyn) [inline docs/upgrade.html]:
```suggestion
        A fix is given in <a href="https://issues.apache.org/jira/browse/ZOOKEEPER-3056">ZOOKEEPER-3056</a>, which is to set the <code>snapshot.trust.empty=true</code>
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a couple of suggestions.

**Review by rondagostino — COMMENTED** [review]
LGTM

**Pull request merged**